src/pip/_internal/cli/cmdoptions.py

```diff
@@ -99,6 +101,135 @@ def check_dist_restriction(options: Values, check_target: bool = False) -> None:
             )
 
 
+def validate_platform_options(platform_options: List[str]) -> None:
+    """
+    Determine if platform options follow standard structures provided
+    in PEPs 425, 513, 571, 599, and 600
+
+    :param options: optparse options as a read-only Values object
+    """
+
+    def is_macos_arch(suffix: str) -> bool:
+        osx_re = re.compile(r"(?P<major>\d+)_(?P<minor>\d+)_(?P<arch>.+)")
+        match = osx_re.fullmatch(suffix)
+        if match:
+            major, minor, arch = (
+                int(match.group("major")),
+                int(match.group("minor")),
+                match.group("arch"),
+            )
+
+            if (major, minor) < (10, 0) or arch not in ["arm64", "intel", "x86_64"]:
+                return False
+        else:
+            return False
+        return True
```

Comment:
Fail fast makes it easier to read.
```suggestion
        if not match:
            return False

        major, minor, arch = (
            int(match.group("major")),
            int(match.group("minor")),
            match.group("arch")
        )
        return (major, minor) >= (10, 0) or arch in {"arm64", "intel", "x86_64"}:
```